### PR TITLE
playground: reason syntax and react ppx

### PR DESCRIPTION
**Pull request by jchavarri**
Based on #538, continues the work from there to add:

- [x] Integration with reason syntax
- [x] reactjs ppx
- [x] removes special handling for stdlib, so packages like `melange.belt` are required as if they were in node_modules
- [x] investigate issue with reason parser choking on `//` input
- [x] issue with errors without locations (https://github.com/melange-re/melange/pull/602#issuecomment-1573968523)

### CODE REVIEW

**Comment by jchavarri** [comment]
@anmonteiro I am running into some issue with errors from ppxlib, where the locations are missing for some reason.

Repro:

```
$ make dev
$ find _build/default/jscomp/stdlib-412/.stdlib.objs/melange -name "*.cmi" -or -name "*.cmj" | xargs js_of_ocaml build-fs -o stdlib.js
$ make playground-dev
$ node
Welcome to Node.js v18.16.0.
Type ".help" for more information.
> require('./_build/default/jscomp/main/jsoo_main.bc.js'); require('./stdlib.js');
{}
> ocaml.compile("let t = [%bs.obj 2]");
{ js_error_msg: 'Expect a record expression here' }
```

The problem seems to be that `Location.error_of_exn` returns `None` for this error, do you have any clues? I tried removing the code added in https://github.com/melange-re/melange/pull/590, but nothing changed.

File: bin/jsoo_common.ml
Comment (anmonteiro) [inline bin/jsoo_common.ml:89]: isn't this https://github.com/reasonml/reason/issues/2350, which was fixed?

Comment (jchavarri) [reply bin/jsoo_common.ml:89]: It looks like. I have no idea why it's reappearing now, maybe because in playground case we don't have EOF?

**Comment by anmonteiro** [comment]
mystery solved https://github.com/melange-re/melange/pull/602/commits/1b89c2f50b0b8dadef732a4578e49934502f9214

**Review by anmonteiro — COMMENTED** [review]
I'd like to depend on Reason 3.9 before we merge this (see https://github.com/ocaml/opam-repository/pull/23945).

That release ports Reason to ppxlib and breaks the current interface file signature.

**Comment by jchavarri** [comment]
> mystery solved https://github.com/melange-re/melange/commit/1b89c2f50b0b8dadef732a4578e49934502f9214

Thanks!

> I'd like to depend on Reason 3.9 before we merge this (see https://github.com/ocaml/opam-repository/pull/23945).

Sounds good.

File: bin/jsoo_main.ml
Comment (anmonteiro) [inline bin/jsoo_main.ml:111]: perhaps we should call this `melange`.

feel free to do in a future PR, I don't want to unblock this one further

**Review by anmonteiro — APPROVED** [review]

**Pull request merged**
